### PR TITLE
[ATO 490] Rasa train doesn't throw an error if form in slot mappings active loop doesn't exist

### PR DESCRIPTION
**Pull request by Urkem**
**Proposed changes**:
- Implement [ATO-490](https://rasahq.atlassian.net/browse/ATO-490) by adding flags `--validate-before-training`, `--fail-on-validation-warnings` and `--validation-max-history` to `rasa train` to bring it inline with `rasa data validate`

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

[ATO-490]: https://rasahq.atlassian.net/browse/ATO-490?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/cli/test_utils.py
Comment (ancalita) [inline tests/cli/test_utils.py:344]: Could we please use the argument names in the function call to improve readability? 🙏🏻

File: tests/cli/test_utils.py
Comment (ancalita) [inline tests/cli/test_utils.py:372]: Same here, please use argument names.

File: tests/cli/test_rasa_train.py
Comment (ancalita) [inline tests/cli/test_rasa_train.py:539]: We should also have a `test_train_validation_max_history` too.

File: changelog/12324.improvement.md
Comment (ancalita) [inline changelog/12324.improvement.md:1]: I'd add a bit more detail here to justify why this improvement was required/ requested (to align validation issues detected by rasa data validate with rasa train).

**Review by ancalita — COMMENTED** [review]
Looks great 🎉 Left some suggestions for improvements.
I'd also recommend outlining this solution to Product and CSE to get their stamp of approval (for the overall solution of giving flexibility to run validation checks via additional flags).

File: tests/cli/test_rasa_train.py
Comment (ancalita) [inline tests/cli/test_rasa_train.py:588]: Shouldn't the exit code be 1 if there was an error found? 🤔

**Review by ancalita — COMMENTED** [review]
LGTM 👍🏻 All my previous comments have been addressed, I only asked a clarifying question on a unit test.

File: tests/cli/test_rasa_train.py
Comment (Urkem) [reply tests/cli/test_rasa_train.py:588]: By default these validations are warnings, training can complete with these warnings unless `--fail-on-validation-warnings` is active. This one is testing that max-history flag works the `test_train_validation_max_history_2` is the same test but it found no conflicts in stories because it goes deeper  `--validation-max-history = 2`.

Comment (ancalita) [reply tests/cli/test_rasa_train.py:588]: Got it, thanks!

File: docs/docs/command-line-interface.mdx
Comment (m-vdb) [inline docs/docs/command-line-interface.mdx:140]:
```suggestion
The `--validation-max-history` is analogous to the `--max-history` argument of `rasa data validate`. 
```

**Comment by m-vdb** [comment]
@ancalita it sounds like this is ready to get merged; if so, can you approve the PR?

File: docs/docs/command-line-interface.mdx
Comment (ancalita) [reply docs/docs/command-line-interface.mdx:140]: ➕

File: rasa/cli/interactive.py
Comment (ancalita) [inline rasa/cli/interactive.py:108]: Shouldn't this be `False` by default?

**Review by ancalita — APPROVED** [review]
Looks great 💯 Only two tiny comments that should be addressed before merging.

File: rasa/cli/interactive.py
Comment (Urkem) [reply rasa/cli/interactive.py:108]: I didn't want to change the default behaviour for `rasa interactive` to validate by default only `rasa train` since thats what the issue in the ticket was about.

Comment (ancalita) [reply rasa/cli/interactive.py:108]: Oh I see, all clear now! Cheers.

**Pull request merged**
